pkg/controller/iotconfig/jdbc_device_connection.go

```diff
@@ -0,0 +1,261 @@
+			return err
+		}
+
+		// volume mounts
+
+		install.ApplyVolumeMountSimple(container, "config", "/etc/config", true)
+		install.ApplyVolumeMountSimple(container, "tls", "/etc/tls", true)
+
+		// extensions
+		ext.MapExtensionVolume(container)
+
+		// apply container options
+
+		if service.JDBC != nil {
+			applyContainerConfig(container, service.JDBC.Container)
+		}
+
+		// return
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	// extension containers
+
+	if service.JDBC.Server.External != nil {
+		if err := ext.AddExtensionContainers(service.JDBC.Server.External.Extensions, &deployment.Spec.Template.Spec, "ext-"); err != nil {
+			return err
+		}
+	} else {
+		// we don't support any init containers
+		deployment.Spec.Template.Spec.InitContainers = nil
+	}
+
+	// tracing
+
+	SetupTracing(config, deployment, tracingContainer)
+
+	// volumes
+
+	install.ApplyConfigMapVolume(&deployment.Spec.Template.Spec, "config", nameDeviceConnection+"-config")
+	ext.AddExtensionVolume(&deployment.Spec.Template.Spec)
+
+	// inter service secrets
+
+	if err := ApplyInterServiceForDeployment(config, deployment, nameDeviceConnection); err != nil {
+		return err
+	}
+
+	// return
+
+	return nil
+}
+
+func ExternalJdbcConnectionConnections(config *iotv1alpha1.IoTConfig) (*iotv1alpha1.JdbcConnectionInformation, error) {
+	if config.Spec.ServicesConfig.DeviceConnection.JDBC.Server.External != nil {
+		return &config.Spec.ServicesConfig.DeviceConnection.JDBC.Server.External.JdbcConnectionInformation, nil
+	} else {
+		return nil, util.NewConfigurationError("illegal device connection configuration")
+	}
+}
+
+func (r *ReconcileIoTConfig) reconcileJdbcDeviceConnectionConfigMap(config *iotv1alpha1.IoTConfig, configMap *corev1.ConfigMap, change *cchange.ConfigChangeRecorder) error {
+
+	install.ApplyDefaultLabels(&configMap.ObjectMeta, "iot", configMap.Name)
+
+	if configMap.Data == nil {
+		configMap.Data = make(map[string]string)
+	}
+
+	if configMap.Data["logback-spring.xml"] == "" {
+		configMap.Data["logback-spring.xml"] = DefaultLogbackConfig
+	}
+
+	deviceInformation, err := ExternalJdbcConnectionConnections(config)
+	if err != nil {
+		return err
+	}
+
+	// app config sections
+
+	app := map[string]interface{}{
+		"hono": map[string]interface{}{
+			"auth": map[string]interface{}{
+				"port":             5671,
+				"keyPath":          "/etc/tls/tls.key",
+				"certPath":         "/etc/tls/tls.crt",
+				"keyFormat":        "PEM",
+				"trustStorePath":   "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt",
+				"trustStoreFormat": "PEM",
+			},
+		},
+		"enmasse": map[string]interface{}{
+			"iot": map[string]interface{}{
+				"app": map[string]interface{}{
+					"maxInstances": 1,
+				},
+				"vertx": map[string]interface{}{
+					"preferNative": true,
+				},
+				"healthCheck": map[string]interface{}{
+					"insecurePortBindAddress": "0.0.0.0",
```

Comment:
Should this be 127.0.0.1?

Reply:
This needs to be accessed by the health probe and the prometheus scraper. From what I remember that didn't work when it was bound to 127.0.0.1.